src/Plugin/BulletinBase.php

```diff
@@ -0,0 +1,72 @@
+<?php
+
+namespace Drupal\govdelivery_bulletins\Plugin;
+
+use Drupal\Core\Plugin\ContainerFactoryPluginInterface;
+use Drupal\Core\Queue\QueueWorkerBase;
+
+/**
+ * Provides base functionality for the NodePublish Queue Workers.
+ */
+class BulletinBase extends QueueWorkerBase implements ContainerFactoryPluginInterface {
+
+  /**
+   * BulletinBase constructor.
+   */
+  public function __construct() {
+
+  }
+
+  /**
+   * Processes the bulletins in the queue.
+   *
+   * @param DateTime $end_time
+   *   The timestamp (optional) to use for grabbing items created prior.
+   */
+  public function processQueue($end_time = NULL) {
+    // Call our queue service.
+    $queue_factory = \Drupal::service('queue');
+    $queue_manager = \Drupal::service('plugin.manager.queue_worker');
+
+    // Call bulletins service, and create an instance for processing.
+    $queue = $queue_factory->get('govdelivery_bulletins');
+    $queue_worker = $queue_manager->createInstance('govdelivery_bulletins');
+
+    // Get the number of items.
+    $number_of_queue = $queue->numberOfItems();
+
+    for ($i = 0; $i < $number_of_queue; $i++) {
+      // Get a queued item.
+      $item = $queue->claimItem();
+      if (($end_time && $item->created < $end_time) || empty($end_time)) {
+        // Call the send service.
+        $send = self::send($item->data);
```

Comment:
```suggestion
        $send = $this->send($item->data);
```
Since send should be a private function (see farther down).

Reply:
:+1: